WalletWasabi.Gui/MainWindow.xaml.cs

```diff
@@ -44,32 +50,35 @@ public MainWindow()
 		}
 
 		private Global Global { get; }
+		public TerminateService TerminateService { get; }
 
 		private void InitializeComponent()
 		{
 			AvaloniaXamlLoader.Load(this);
 		}
 
 		private int _closingState;
+		private bool IsClosed => _closingState == ClosingStateClosed;
```

Comment:
Missing that `Interlocked` thingy.

Reply:
According to this, it is not necessary, in addition there is no InterLocked.Read for int. 

https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.read?view=netcore-3.1

Reply:
I believe you are wrong. When you have a variable that is accessed by multiple threads then you need to make sure that each thread gets correct value. There is no `Interlocked.Read` for int because you will get it atomically anyway on all systems (there are no 16-bit systems anyway). But you can still get stale value (cached value, not up-to-date value), that's why in `WasabiSynchronizer`, state is represented using `long` and not `int` and each read&write is done using Interlocked.

Reply:
> https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.read?view=netcore-3.1

The remark only says that when you have `long` value, the two 32-bit parts may get corrupted if you don't use Interlocked.Read on 32-bit systems.

Reply:
See https://stackoverflow.com/questions/1186515/interlocked-and-volatile/1186562#1186562